src/main/java/org/mockito/plugins/InstantiatorProvider2.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) 2018 Mockito contributors
+ * This program is made available under the terms of the MIT License.
+ */
+package org.mockito.plugins;
+
+import org.mockito.creation.instance.Instantiator;
+import org.mockito.mock.MockCreationSettings;
+
+/**
+ * <p>
+ *     Mockito will invoke this interface in order to fetch an instance instantiator provider.
+ * </p>
+ *
```

Comment:
Can you explain somewhere here why deprecated previous InstantiatorProvider and created this class? It's a good dev experience when a concise justification of deprecation/API change is offered in Javadoc (plus link to github ticket).

Please also explain that the old InstantiatorProvider is supported but the new one takes precedence during plugin loading.

Finally, we'd love to have a new paragraph in the main Mockito class explaining the addition of new public API (see example: https://github.com/mockito/mockito/blob/release/2.x/src/main/java/org/mockito/Mockito.java#L100)

Reply:
- I think you already added a comment why this was deprecated
- Added link to #1303
- Added comment about plugin preference if both a InstantiatorProvider and InstantiatorProvider2 are configured
- Created entry 44 in Mockito.java

Reply:
Great!